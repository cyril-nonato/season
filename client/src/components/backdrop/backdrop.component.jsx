import React from 'react'
import * as S from './backdrop.styles';

const Backdrop = ({ click }) => {
  return (
    <S.BackdropContainer data-testid='backdrop' onClick={click} />
  );
}

export default Backdrop;