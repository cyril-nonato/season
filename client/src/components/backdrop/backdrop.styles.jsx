import styled from 'styled-components';

export const BackdropContainer = styled.div`
  position: fixed;
  width: 100vw;
  height: 100vh;
  background-color: transparent;
  z-index: 10;
  top: 0;
  left: 0;
`