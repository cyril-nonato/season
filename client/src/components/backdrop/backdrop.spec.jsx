import React from 'react';
import {render, fireEvent} from '@testing-library/react'
import 'jest-styled-components'

import Backdrop from './backdrop.component'

import * as S from './backdrop.styles'

describe('Backdrop Component Test', () => {
  const props = {
    click: jest.fn()
  }

  test('Should call the click function when clicked', () => {
    const {getByTestId} = render(<Backdrop {...props} />);

    fireEvent.click(getByTestId('backdrop'));

    expect(props.click).toHaveBeenCalled();
  });

  test('Match the snapshot', () => {
    const {asFragment} = render(<S.BackdropContainer />);

    expect(asFragment()).toMatchSnapshot();
  });
});
