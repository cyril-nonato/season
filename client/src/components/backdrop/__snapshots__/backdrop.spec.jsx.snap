// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Component Test Match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  width: 100vw;
  height: 100vh;
  background-color: rgba(255,255,255,.5);
  z-index: 5;
  top: 0;
  left: 0;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
